science/mpich/Portfile

```diff
@@ -378,6 +378,11 @@ if {${subport} eq "mpich-default"} {
         }
     }
 
+    variant f08 description {Support f08 (build with CC=gcc)} {
```

Comment:
Is there some special reason not to make this default (and make it a variant at all instead of having it enabled)?
Any non-default variant means that no dependents using it can pass CI; it also will force end-users to build MPICH from source, and force maintainers to add extra code to check active variants and require Fortran enabled.

Reply:
I’m debating that, but right now the build fails during configure for me with gcc14 (or with +f08) on Sonoma. Not sure when I’ll have time to iron that it.

Reply:
My primary concern is CI. (I have to build from source on powerpc anyway, so a non-default variant does not change anything here.)
Broken CI will mean that nothing which uses modern Fortran with MPICH can be updated: MacPorts does not give commit access, and how I am gonna convince someone with commit access that a port actually builds?

There will be side effects like some dependents will appear broken in port details while they are not. This may discourage end-users from installing a port, it may discourage upstreams to cooperate, if they see that their port is left broken, etc.

Then once it is fixed, non-broken dependents will have to be revbumped, since binaries may be different (a dependent may support both standards, but prefer 2008, so enabling it gonna change the way a port is compiled).

Reply:
Was running into this. https://trac.macports.org/ticket/70331

Will re-assess when I have a chance. I agree just building mpich-default with gcc14 (for C and fortran) may be the easiest way to go.